gammapy/background/models.py

```diff
@@ -71,3 +81,775 @@ def evaluate(self, x, y):
+
+        if format == 'bin_table':
+            self.to_fits_bin_table().writeto(outfile, **write_kwargs)
+        elif format == 'image':
+            self.to_fits_image().writeto(outfile, **write_kwargs)
+        else:
+            raise ValueError("Got format {}, which is not understood."
+                             .format(format))
+
+    @property
+    def image_extent(self):
+        """Image extent `(x_lo, x_hi, y_lo, y_hi)`.
+
+        Returns
+        -------
+        im_extent : `~astropy.coordinates.Angle`
+            array of bins with the image extent
+        """
+        bx = self.detx_bins.to('degree')
+        by = self.dety_bins.to('degree')
+        return Angle([bx[0], bx[-1], by[0], by[-1]])
+
+    @property
+    def spectrum_extent(self):
+        """Spectrum extent `(e_lo, e_hi)`.
+
+        Returns
+        -------
+        spec_extent : `~astropy.units.Quantity`
+            array of bins with the spectrum extent
+        """
+        b = self.energy_bins.to('TeV')
+        return Quantity([b[0], b[-1]])
+
+    @property
+    def image_bin_centers(self):
+        """Image bin centers `(x, y)`.
+
+        Returns
+        -------
+        det_edges_centers : `~astropy.coordinates.Angle`
+            array of bins with the image bin centers [[x_centers], [y_centers]]
+        """
+        detx_edges_low = self.detx_bins[:-1]
+        detx_edges_high = self.detx_bins[1:]
+        detx_edges_centers = (detx_edges_low + detx_edges_high)/2.
+        dety_edges_low = self.dety_bins[:-1]
+        dety_edges_high = self.dety_bins[1:]
+        dety_edges_centers = (dety_edges_low + dety_edges_high)/2.
+        return Angle([detx_edges_centers, dety_edges_centers])
+
+    @property
+    def spectrum_bin_centers(self):
+        """Spectrum bin centers (log center)
+
+        Returns
+        -------
+        energy_bin_centers : `~astropy.units.Quantity`
+            array of bins with the spectrum bin centers
+        """
+        energy_edges_low = self.energy_bins[:-1]
+        energy_edges_high = self.energy_bins[1:]
+        e_lo_tev = energy_edges_low.to('TeV').value
+        e_hi_tev = energy_edges_high.to('TeV').value
+        energy_bin_centers = 10.**(0.5*(np.log10(e_lo_tev*e_hi_tev)))
+        return Quantity(energy_bin_centers, 'TeV')
+
+    @property
+    def det_wcs(self):
+        """WCS object describing the coordinates of the det (X, Y) bins.
+
+        This method gives the correct answer only for linear X, Y binning.
+
+        Returns
+        -------
+        wcs : `~astropy.wcs.WCS`
+            WCS object describing the bin coordinates
+        """
+        wcs = make_linear_wcs_from_bin_edges_arrays(name_x="DETX",
+                                                    name_y="DETY",
+                                                    bins_x=self.detx_bins,
+                                                    bins_y=self.detx_bins)
+        return wcs
+
+    def find_det_bin(self, det):
```

Comment:
I'm not happy with this long function.
It can probably be improved significantly by using the `wcs` object and general utility functions like [gammapy.image.lookup](https://gammapy.readthedocs.org/en/latest/api/gammapy.image.lookup.html) and then for filling [gammapy.image.wcs_histogram2d](https://gammapy.readthedocs.org/en/latest/api/gammapy.image.wcs_histogram2d.html).

If you want to defer the fill / lookup to a future, PR, just add a TODO here that this should be re-discussed and re-written. Otherwise, please try to rewrite this to use `wcs` to find bins.